ios/NetworkModule.swift

```diff
@@ -0,0 +1,98 @@
+import BitmovinPlayer
+
+@objc(NetworkModule)
+public class NetworkModule: NSObject, RCTBridgeModule {
+    // swiftlint:disable:next implicitly_unwrapped_optional
+    @objc public var bridge: RCTBridge!
+
+    /// In-memory mapping from `nativeId`s to `NetworkConfig` instances.
+    private var networkConfigs: Registry<NetworkConfig> = [:]
+    private var preprocessHttpResponseCompletionHandlers: Registry<(_ response: HttpResponse) -> Void> = [:]
+
+    // swiftlint:disable:next implicitly_unwrapped_optional
+    public static func moduleName() -> String! {
+        "NetworkModule"
+    }
+
+    public static func requiresMainQueueSetup() -> Bool {
+        true
+    }
+
+    // swiftlint:disable:next implicitly_unwrapped_optional
+    public var methodQueue: DispatchQueue! {
+        bridge.uiManager.methodQueue
+    }
+
+    @objc
+    func retrieve(_ nativeId: NativeId) -> NetworkConfig? {
+        networkConfigs[nativeId]
+    }
+
+    @objc(initWithConfig:config:)
+    func initWithConfig(_ nativeId: NativeId, config: Any?) {
+        bridge.uiManager.addUIBlock { [weak self] _, _ in
+            guard
+                self?.retrieve(nativeId) == nil,
+                let networkConfig = RCTConvert.networkConfig(config)
+            else {
+                return
+            }
+            self?.networkConfigs[nativeId] = networkConfig
+            self?.initConfigBlocks(nativeId, config)
+        }
+    }
+
+    @objc(destroy:)
+    func destroy(_ nativeId: NativeId) {
+        networkConfigs.removeValue(forKey: nativeId)
+
+        preprocessHttpResponseCompletionHandlers.keys.filter { $0.starts(with: nativeId) }.forEach {
+            preprocessHttpResponseCompletionHandlers.removeValue(forKey: $0)
+        }
+    }
+
+    private func initConfigBlocks(_ nativeId: NativeId, _ config: Any?) {
+        if let json = config as? [String: Any] {
+            initPreprocessHttpResponse(nativeId, networkConfigJson: json)
+        }
+    }
+
+    private func initPreprocessHttpResponse(_ nativeId: NativeId, networkConfigJson: [String: Any]) {
+        guard let networkConfig = retrieve(nativeId) else {
+            return
+        }
+        if networkConfigJson["preprocessHttpResponse"] != nil {
```

Comment:
nit: We could move this statement into the `guard` above

Reply:
Let's do it 🙌 , it's always a nice idea to simplify the code 🙂 
Address in c91309699d564b216c061ede528d66f8a2c6a5e6